HealthMonitoring.Persistence/EndpointStatsRepository.cs

```diff
@@ -40,23 +40,14 @@ public IEnumerable<EndpointStats> GetStatistics(Guid id, int limitDays)
             }
         }
 
-        public void DeleteStatistics(Guid id)
+        public int DeleteStatisticsOlderThan(DateTime date, int maxBatchSize)
         {
             using (var conn = _db.OpenConnection())
             using (var tx = conn.BeginTransaction())
             {
-                conn.Execute("delete from EndpointStats where endpointId=@id", new { id }, tx);
-                tx.Commit();
-            }
-        }
-
-        public void DeleteStatisticsOlderThan(DateTime date)
-        {
-            using (var conn = _db.OpenConnection())
-            using (var tx = conn.BeginTransaction())
-            {
-                conn.Execute("delete from EndpointStats where CheckTimeUtc < @date", new { date }, tx);
+                var count = conn.Execute("delete from EndpointStats where CheckTimeUtc < @date order by CheckTimeUtc limit @maxBatchSize", new { date, maxBatchSize }, tx);
```

Comment:
Instead of deleting all stats at once, only up to maxBatchSize would be deleted to make transaction small and allow other transactions to execute in meantime